packages/web/src/js/Dropdown.ts

```diff
@@ -0,0 +1,203 @@
+import { computePosition, flip, shift, offset, autoUpdate } from '@floating-ui/dom';
+import BaseComponent from './BaseComponent';
+import { enableToggleTrigger } from './utils/ComponentFunctions';
+import EventHandler from './dom/EventHandler';
+import SelectorEngine from './dom/SelectorEngine';
+
+enum placementKeys {
+  'top-start' = 'top-start',
+  'top-end' = 'top-end',
+  'right-start' = 'right-start',
+  'right-end' = 'right-end',
+  'bottom-start' = 'bottom-start',
+  'bottom-end' = 'bottom-end',
+  'left-start' = 'left-start',
+  'left-end' = 'left-end',
+}
+
+interface DropdownStateProps {
+  open: boolean;
+}
+
+interface DropdownOptionsProps {
+  placement: keyof typeof placementKeys;
+  offset: number;
+  padding: number;
+  autoClose: boolean;
+}
+
+type autoUpdateOptionsType = {
+  ancestorScroll?: boolean;
+  ancestorResize?: boolean;
+  elementResize?: boolean;
+  animationFrame?: boolean;
+};
+
+const NAME = 'dropdown';
+const DATA_KEY = `${NAME}`;
+const EVENT_KEY = `.${DATA_KEY}`;
+const EVENT_HIDE = `hide${EVENT_KEY}`;
+const EVENT_HIDDEN = `hidden${EVENT_KEY}`;
+const EVENT_SHOW = `show${EVENT_KEY}`;
+const EVENT_SHOWN = `shown${EVENT_KEY}`;
+const ARIA_EXPANDED_ATTRIBUTE = 'aria-expanded';
+const ARIA_CONTROLS_ATTRIBUTE = 'aria-controls';
+const CLASSNAME_DROPDOWN = 'dropdown';
+export const CLASSNAME_EXPANDED = 'is-expanded';
+export const CLASSNAME_OPEN = 'is-open';
+
+const clickOutsideElement = (target: Element, event: Event) => !event.composedPath().includes(target);
+
+class Dropdown extends BaseComponent {
+  target: HTMLElement | null | undefined;
+  anchor: HTMLElement | null | undefined;
+  state: DropdownStateProps;
+  options: DropdownOptionsProps;
+  autoUpdateOptions: autoUpdateOptionsType;
+
+  constructor(element: SpiritElement) {
+    super(element);
+    this.target = this.findTargetElement();
+    this.anchor = this.findAnchorElement();
```

Comment:
Is the reference element necessary? Wouldn't it be easier to just use the parent element, eg. the dropdown and the trigger element should be wrapped into the div? Also @adamkudrna and @crishpeen?

Reply:
I just following BS with this: https://getbootstrap.com/docs/5.2/components/dropdowns/#options